### ISSUE TITLE
Doubles sometimes come back as floats

### ISSUE THREAD

**Issue by cgore** [edma2/clojure-msgpack#13]
As of the head of master at this time (ca1d7cb), sometimes encoding a double will decode as a float.
The simplest example of this is as such:
```clojure
msgpack.core> (class (unpack (pack (double 1/3))))
;; => java.lang.Float
```
I would have expected this to come back as a double.

**Comment by edma2** [comment]
Nice catch - that's a pretty stupid bug. The intention was to serialize doubles as floats if it can be done without losing precision. The MessagePack spec recommends to serialize data in the most compact representation possible. As such, I'm not sure that preserving doubles as doubles is what I want if the value can be safely serialized as a float. There's some ongoing discussion about how MessagePack should treat doubles vs floats: https://github.com/msgpack/msgpack/pull/200#issuecomment-112664938

**Comment by cgore** [comment]
Maybe a parameter or switch then, to prefer floats/doubles?  We specifically want the higher precision for our application, but it makes sense that others might be more concerned with the message size.

**Comment by edma2** [comment]
What about just casting all floats you unpack as doubles?

**Comment by edma2** [comment]
Okay, let's preserve precision. See https://github.com/edma2/clojure-msgpack/pull/15

**Comment by edma2** [comment]
Fixed in #15
